### ISSUE TITLE
Print Preview Doesn't Work

### ISSUE THREAD

**Issue by nedrichards** [flathub/org.gimp.GIMP#7]
Print Preview tries to shell out to evince, which doesn't work from within a Flatpak.

> (print:17): Gtk-WARNING **: 19:15:06.940: Error launching preview Failed to execute child process “evince” (No such file or directory)

**Comment by Jehan** [comment]
Well I think GIMP is just using GTK+ code here (to be verified).

@alexlarsson: Is there any portal for this in flatpak?

**Comment by Jehan** [comment]
Just for the record, I can't handle all bugs related to flatpak integration by myself. I am already overwhelmed and need to recenter my activities to the features we actually need for our activity (which is animation film!). So if anyone wants to see this fixed, I would suggest making a patch to GIMP implementing whatever portal is needed and post it on our main tracker: https://gitlab.gnome.org/GNOME/gimp/

Of course such implementation will need to not break non-flatpak GIMP (I don't know if this is one of these, but we have other pieces of code using portals where we basically use the portals if they exist, and fallback to old-style code otherwise).

**Comment by hfiguiere** [comment]
I think there is no fix with gtk2 - Inkscape had similar issues prior to 1.0. We can check when there will be a 2.99 release.

**Comment by Jehan** [comment]
Has anyone tested our GIMP 2.99.2 dev flatpak with print and could tell if it works?

https://www.gimp.org/downloads/devel/

**Comment by wjt** [comment]
Yes, it works. org.gimp.GIMP from flathub-beta opens the print portal, and it has a Preview button which works.

(This led me to discover that Endless OS can't remove Evince from the ostree in favour of the Flatpak version because this will break… print preview!)
